sympy/functions/elementary/integers.py

```diff
@@ -216,6 +217,12 @@ def __lt__(self, other):
 
         return Lt(self, other, evaluate=False)
 
+@dispatch(floor, Basic)
+def _eval_Eq(lhs, rhs): # noqa:F811
```

Comment:
I don't think a `floor` could ever compare equal to a non-`Expr` so this should probably by `(floor, Expr)`.